### PR TITLE
Refactor GeoTIFF Validator

### PR DESCRIPTION
**Pull request by JakobMiksch**
- split GeoTIFF validator in multiple files 
- add basic tests
- removes AJV, because input validation is now done within the validator-functions
- followup-PR to fix tests https://github.com/klips-project/rabbitmq-worker/pull/123

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -18,6 +18,8 @@ jobs:
         uses: actions/setup-node@v2.1.5
         with:
           node-version: '16'
+      - run: sudo apt update && sudo apt install gdal-bin -y
+      - run: gdalinfo --version
```
Comment (hblitza) [inline .github/workflows/test.yml:22]: good idea

File: src/geotiff-validator/index.js [hunk 2]
```diff
@@ -1,227 +1,26 @@
...
 import path from 'path';
 
-import { transformExtent } from 'ol/proj.js';
-import { boundingExtent, containsExtent } from 'ol/extent.js';
-import { register } from 'ol/proj/proj4.js';
-import proj4 from 'proj4';
-
-import { initialize, log, debugLog } from '../workerTemplate.js';
+import { initialize, log } from '../workerTemplate.js';
+import { createGeotiffValidationFun } from './worker.js';
```
Comment (hblitza) [inline src/geotiff-validator/index.js:5]: Fun? ah function

File: src/geotiff-validator/package.json [hunk 3]
```diff
@@ -33,4 +31,4 @@
     "url": "https://github.com/klips-project/rabbitmq-worker/issues"
   },
   "homepage": "https://github.com/klips-project/rabbitmq-worker#readme"
-}
+}
```
Comment (hblitza) [inline src/geotiff-validator/package.json:34]: missing new line?

**Comment by hblitza** [comment]
LGTM in general (untested) Makes sense to split the worker. Maybe: Update readme if/where needed.

**Review by hblitza — APPROVED** [review]

File: src/geotiff-validator/package.json (hunk 3 above)
Comment (JakobMiksch) [reply src/geotiff-validator/package.json]: fixed

File: .github/workflows/test.yml (hunk 1 above)
Comment (JakobMiksch) [reply .github/workflows/test.yml]: I actually move it to the follow up PR, because it is not needed yet

**Comment by hblitza** [comment]
Thanks, please merge!

**Pull request merged**